File: src/main/java/com/capscrum/school/erp/dataaccessor/service/AuthService.java
package com.capscrum.school.erp.dataaccessor.service;

import com.capscrum.school.erp.dataaccessor.dto.AuthResponse;
import com.capscrum.school.erp.dataaccessor.dto.LoginRequest;
import com.capscrum.school.erp.dataaccessor.dto.RegisterRequest;
import com.capscrum.school.erp.dataaccessor.dto.UserDto;

public interface AuthService {

    AuthResponse login(LoginRequest loginRequest);

    AuthResponse register(RegisterRequest registerRequest);

    UserDto getCurrentUser(String username);
}

File: frontend/src/api/authService.ts
import api from './axios';
import type { JwtAuthResponse, LoginRequest, RegisterRequest } from '../types/domain';

export const authService = {
  login: async (data: LoginRequest): Promise<JwtAuthResponse> => {
    const response = await api.post<JwtAuthResponse>('/api/auth/login', data);
    return response.data;
  },
  
  register: async (data: RegisterRequest): Promise<JwtAuthResponse> => {
    const response = await api.post<JwtAuthResponse>('/api/auth/register', data);
    return response.data;
  },

  getCurrentUser: async () => {
    const response = await api.get('/api/auth/me');
    return response.data;
  },
};
